pkg/ccl/changefeedccl/sink_kafka.go

```diff
@@ -400,6 +400,10 @@ func (s *kafkaSink) startInflightMessage(ctx context.Context) error {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
+	if s.mu.flushErr != nil {
+		return s.mu.flushErr
+	}
```

Comment:
I don't know if this is needed or not since we didn't have any check like this before my PR either and `flushErr` is being assigned in the same way, but I'm also not sure how we didn't have any issues with the original behaviour without having this kind of check.  It might be that the sarama producer never allowed future messages to that partition but in the code I can't find out where it would do that after its given up on retrying.

Reply:
flushErr -- that handling -- or maybe even the bug -- existed since forever.
It was really bad that the error would not be surfaced until the next Flush requests -- thus, we even had a possibility
of emitting out of order keys (i.e. k@t1 fails, we don't notice it for a while, we emit k@t2 -- that succeeds, then we flush -- notice error, and restart, and re-emit k@t1).

This particular issue made matters worse since it appears to drop entire batch of messages, and (for reasons yet unknown), when running with initial scan only, the flushErr is never propagated.

I don't think sarama gives up on anything -- it just returned an error, we recorded it (as flushErr).
It's not clear to me what happens then -- we should investigate in a lot more detail later.

So, yes, this check is newly added/needed -- since it's the only way I can see (without making drastic changes) on how to propagate error sooner.